### PR TITLE
Multiple targets warning

### LINKED ISSUE #5144: Max 1 target variable per domain
- [ ] **What's your use case?**

I'm a new user, and I'm picking up data mining as I go. I don't know what meta attributes are, and I'm not quite sure what a target variable is. Can I set multiple target variables? 
Upon doing so and connecting a model, a `Data has no target variable` error appears. Confusing. Rank displays `Data does not have a (single) target variable`, slightly better.

- [ ] **What's your proposed solution?**

Limit the number of target variables to one. If a new target variable is set, set the previous one to the role it previously played. Make sure this behavior is reflected in the file loading widget's domain editor and Select Columns. Make the Select Columns box 'target variable' box smaller, to fit one variable.

- [ ] **Are there any alternative solutions?**

Better, unified error messages.

### PR DESCRIPTION
**Pull request by janezd**
##### Issue

Fixes #5144.

##### Description of changes

- File widget and Select Columns now show warnings when outputting data with multiple targets.
- Widgets that showed potentially misleading error (learners and freeviz) now explicitly state that the data has to have a **single** target.

##### Includes
- [X] Code changes
- [ ] Tests

### CODE REVIEW

File: Orange/widgets/utils/owlearnerwidget.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def set_data(self, data):
         self.set_input_summary()
 
         if data is not None and data.domain.class_var is None:
-            self.Error.data_error("Data has no target variable.\n"
+            self.Error.data_error("Data must have a (single) target variable.\n"
```
Comment (irgolic) [inline Orange/widgets/utils/owlearnerwidget.py:139]: Can we be more explicit here? As in, a different message if there's no target vs if there are too many targets?

Something like `'Data must have one, not %d target variables.\n'`

Comment (ajdapretnar) [reply Orange/widgets/utils/owlearnerwidget.py:139]: Agree. I would put two different errors here. They require similar, but different actions (add / remove).

File: Orange/widgets/utils/owlearnerwidget.py [hunk 2]
```diff
@@ -136,8 +136,14 @@ def set_data(self, data):
...
-            self.Error.data_error("Data has no target variable.\n"
-                                  "You can set a target variable with the Select Columns widget.")
+            if data.domain.class_vars:
+                self.Error.data_error(
+                    "Data contains multiple target variables.\n"
+                    "You can select a single one with Select Columns widget.")
+            else:
+                self.Error.data_error(
+                    "Data has no target variable.\n"
+                    "You can select one with Select Columns widget.")
```
Comment (irgolic) [inline Orange/widgets/utils/owlearnerwidget.py:146]: the Select Columns widget

File: Orange/widgets/utils/owlearnerwidget.py [hunk 3]
```diff
@@ -136,8 +136,14 @@ def set_data(self, data):
         self.set_input_summary()
 
         if data is not None and data.domain.class_var is None:
-            self.Error.data_error("Data has no target variable.\n"
-                                  "You can set a target variable with the Select Columns widget.")
+            if data.domain.class_vars:
+                self.Error.data_error(
+                    "Data contains multiple target variables.\n"
+                    "You can select a single one with Select Columns widget.")
```
Comment (irgolic) [inline Orange/widgets/utils/owlearnerwidget.py:142]: the Select Columns widget

File: Orange/widgets/visualize/owfreeviz.py [hunk 4]
```diff
@@ -139,14 +139,16 @@ class OWFreeViz(OWAnchorProjectionWidget, ConcurrentWidgetMixin):
     graph = settings.SettingProvider(OWFreeVizGraph)
 
     class Error(OWAnchorProjectionWidget.Error):
-        no_class_var = widget.Msg("Data has no target variable")
+        no_class_var = widget.Msg("Data must have a target variable.")
+        multiple_class_vars = widget.Msg(
+            "Data must have just a single target variable.")
         not_enough_class_vars = widget.Msg(
-            "Target variable is not at least binary")
+            "Target variable is not at least binary.")
```
Comment (irgolic) [inline Orange/widgets/visualize/owfreeviz.py:146]: Target variable must have at least two unique values?

File: Orange/widgets/utils/owlearnerwidget.py (hunk 2 above)
Comment (janezd) [reply Orange/widgets/utils/owlearnerwidget.py]: I intentionally removed `the` to make it shorter. But OK, I added it back.

Comment (irgolic) [reply Orange/widgets/utils/owlearnerwidget.py]: Honestly, I didn't open the widget and check. Generally, I'm not too worried about the message length, they're visible on hover/click.

File: Orange/widgets/visualize/owfreeviz.py [hunk 5]
```diff
@@ -139,14 +139,16 @@ class OWFreeViz(OWAnchorProjectionWidget, ConcurrentWidgetMixin):
     graph = settings.SettingProvider(OWFreeVizGraph)
 
     class Error(OWAnchorProjectionWidget.Error):
-        no_class_var = widget.Msg("Data has no target variable")
+        no_class_var = widget.Msg("Data must have a target variable.")
+        multiple_class_vars = widget.Msg(
+            "Data must have just a single target variable.")
```
Comment (ajdapretnar) [inline Orange/widgets/visualize/owfreeviz.py:144]: "Just" is likely redundant. I prefer:
- "Data must have a single target variable."
- "Data must have only one target variable."

File: Orange/widgets/utils/owlearnerwidget.py [hunk 6]
```diff
@@ -136,8 +136,14 @@ def set_data(self, data):
         self.set_input_summary()
 
         if data is not None and data.domain.class_var is None:
-            self.Error.data_error("Data has no target variable.\n"
-                                  "You can set a target variable with the Select Columns widget.")
+            if data.domain.class_vars:
+                self.Error.data_error(
+                    "Data contains multiple target variables.\n"
+                    "You can select a single one with the Select Columns widget.")
```
Comment (ajdapretnar) [inline Orange/widgets/utils/owlearnerwidget.py:142]: "You can select a single one with the Select Columns widget." --> "Select a single one with Select Columns." (shorter?)

File: Orange/widgets/utils/owlearnerwidget.py [hunk 7]
```diff
@@ -136,8 +136,14 @@ def set_data(self, data):
...
+            else:
+                self.Error.data_error(
+                    "Data has no target variable.\n"
+                    "You can select one with the Select Columns widget.")
```
Comment (ajdapretnar) [inline Orange/widgets/utils/owlearnerwidget.py:146]: Same as above.

File: Orange/widgets/utils/owlearnerwidget.py (hunk 6 above)
Comment (janezd) [reply Orange/widgets/utils/owlearnerwidget.py]: "You can" was meant to sound as suggestion, just one of possible options. It obviously didn't, so I removed it.

File: Orange/widgets/visualize/owfreeviz.py (hunk 5 above)
Comment (janezd) [reply Orange/widgets/visualize/owfreeviz.py]: OK. I chose the first one, the shorter.

File: Orange/widgets/utils/owlearnerwidget.py (hunk 6 above)
Comment (ajdapretnar) [reply Orange/widgets/utils/owlearnerwidget.py]: I started from the position that I would like to use the learner, so I *have* to have just a single target variable. 😁

**Pull request merged**